sormas-ui/src/main/java/de/symeda/sormas/ui/location/LocationEditForm.java

```diff
@@ -286,6 +286,35 @@ protected void addFields() {
 				facilityDetails.setRequired(false);
 				facilityDetails.clear();
 			}
+
+			// Fill in the address fields based on the selected facility
+			// We don't want the location form to automatically change even if the facility's address is updated later 
+			// on, so we only trigger it upon a manual change of the facility field
+			// We use isAttached() to avoid the fuss when initializing the form, it may seems a bit hacky, but it is
+			// necessary because isModified() will still return true for a short duration even if we keep the very same 
+			// value because of this field dependencies to other fields and the way updateEnumValues works
+			if (facility.isAttached()) {
+				if (facility.getValue() != null) {
+					FacilityDto facilityDto = FacadeProvider.getFacilityFacade().getByUuid(((FacilityReferenceDto) facility.getValue()).getUuid());
+					cityField.setValue(facilityDto.getCity());
+					postalCodeField.setValue(facilityDto.getPostalCode());
+					streetField.setValue(facilityDto.getStreet());
+					houseNumberField.setValue(facilityDto.getHouseNumber());
+					additionalInformationField.setValue(facilityDto.getAdditionalInformation());
+					areaType.setValue(facilityDto.getAreaType());
+					tfLatitude.setConvertedValue(facilityDto.getLatitude());
+					tfLongitude.setConvertedValue(facilityDto.getLongitude());
+				} else {
+					cityField.clear();
```

Comment:
Just as above, blindly clearing these fields is not a good idea. I actually think we should never do that because even if the user removes the facility from the form, the location details might still be relevant (e.g. it's likely that the city is still the same).